Running method with concurrency gives c error: pointer expected

<!-- It is advisable to update all relevant modules using `v outdated` and `v install` -->

<details>
<summary>V version: V 0.4.9 3c88926, press to see full `v doctor` output</summary>

|V full version      |V 0.4.9 5c65e58.3c88926
|:-------------------|:-------------------
|OS                  |windows, Microsoft Windows 11 Pro 26100 64-bit
|Processor           |24 cpus, 64bit, little endian, AMD Ryzen 9 3900X 12-Core Processor  
|Memory              |16.1GB/31.9GB
|                    |
|V executable        |C:\v\v.exe
|V last modified time|2025-02-23 15:50:09
|                    |
|V home dir          |OK, value: C:\v
|VMODULES            |OK, value: C:\Users\info\.vmodules
|VTMP                |OK, value: C:\Users\info\AppData\Local\Temp\v_0
|Current working dir |OK, value: D:\vbug2
|                    |
|Git version         |git version 2.42.0.windows.2
|V git status        |weekly.2025.08-30-g3c88926f-dirty
|.git/config present |true
|                    |
|cc version          |N/A
|gcc version         |N/A
|clang version       |clang version 17.0.1
|msvc version        |N/A
|tcc version         |tcc version 0.9.27 (x86_64 Windows)
|tcc git status      |thirdparty-windows-amd64 b425ac82
|emcc version        |N/A
|glibc version       |ldd (cygwin) 3.4.7

</details>

**What did you do?**
`./v -g -o vdbg cmd/v && ./vdbg  main.v && D:\vbug2\main.exe`
```v
module main



@[heap]
interface IGameObject {
mut:
	name     string
	parent   ?&IGameObject
	children []&IGameObject
	advance()
}

@[heap]
struct GameObject implements IGameObject {
mut:
	name     string
	parent   ?&IGameObject
	children []&IGameObject
}

struct Ship implements IGameObject {
	GameObject
	speed f32
}

fn (mut gameobject GameObject) advance() {
	for mut child in gameobject.children {
		go child.advance()
	}
}

fn main() {
	mut ship := &Ship{
		name: 'ship'
	}
	ship.advance()
}

```

**What did you see?**
```
================== C compilation error (from tcc): ==============
cc: C:/Users/info/AppData/Local/Temp/v_0/main.01JMSVTNC5B0THKVJY24F34MMC.tmp.c:532: warning: WINVER redefined
cc: C:/Users/info/AppData/Local/Temp/v_0/main.01JMSVTNC5B0THKVJY24F34MMC.tmp.c:2054: error: pointer expected
=================================================================
(You can pass `-cg`, or `-show-c-output` as well, to print all the C error messages).
builder error: 
==================
C error found. It should never happen, when compiling pure V code.
This is a V compiler bug, please report it using `v bug file.v`,
or goto https://github.com/vlang/v/issues/new/choose .
You can also use #help on Discord: https://discord.gg/vlang .
```

**What did you expect to see?**

no c error



> [!NOTE]
> You can use the 👍 reaction to increase the issue's priority for developers.
>
> Please note that only the 👍 reaction to the issue itself counts as a vote.
> Other reactions and those to comments will not be taken into account.

--- a/vlib/v/gen/c/spawn_and_go.v
+++ b/vlib/v/gen/c/spawn_and_go.v
@@ -106,9 +106,9 @@ fn (mut g Gen) spawn_and_go_expr(node ast.SpawnExpr, mode SpawnGoMode) {
 	}
 	if expr.is_method {
 		g.write('${arg_tmp_var}${dot}arg0 = ')
-		if expr.left is ast.Ident && expr.left.obj.typ.is_ptr()
+		if mut expr.left is ast.Ident && expr.left.obj.typ.is_ptr()
 			&& !node.call_expr.receiver_type.is_ptr() {
-			g.write('*')
+			g.write('*'.repeat(expr.left.obj.typ.nr_muls()))
 		}
 		g.expr(expr.left)
 		g.writeln(';')
@@ -271,10 +271,10 @@ fn (mut g Gen) spawn_and_go_expr(node ast.SpawnExpr, mode SpawnGoMode) {
 				receiver_type_name := util.no_dots(rec_cc_type)
 				g.gowrappers.write_string2('${c_name(receiver_type_name)}_name_table[',
 					'arg->arg0')
-				idot := if expr.left_type.is_ptr() { '->' } else { '.' }
+				dot_or_ptr := g.dot_or_ptr(unwrapped_rec_type)
 				mname := c_name(expr.name)
-				g.gowrappers.write_string2('${idot}_typ]._method_${mname}(', 'arg->arg0')
-				g.gowrappers.write_string('${idot}_object')
+				g.gowrappers.write_string2('${dot_or_ptr}_typ]._method_${mname}(', 'arg->arg0')
+				g.gowrappers.write_string('${dot_or_ptr}_object')
 			} else if typ_sym.kind == .struct && expr.is_field {
 				g.gowrappers.write_string('arg->arg0')
 				idot := if expr.left_type.is_ptr() { '->' } else { '.' }

--- a/vlib/v/tests/concurrency/mut_receiver_gowrapper_test.v
+++ b/vlib/v/tests/concurrency/mut_receiver_gowrapper_test.v
@@ -1,0 +1,37 @@
+module main
+
+@[heap]
+interface IGameObject {
+mut:
+	name     string
+	parent   ?&IGameObject
+	children []&IGameObject
+	advance()
+}
+
+@[heap]
+struct GameObject implements IGameObject {
+mut:
+	name     string
+	parent   ?&IGameObject
+	children []&IGameObject
+}
+
+struct Ship implements IGameObject {
+	GameObject
+	speed f32
+}
+
+fn (mut gameobject GameObject) advance() {
+	for mut child in gameobject.children {
+		go child.advance()
+	}
+}
+
+fn test_main() {
+	mut ship := &Ship{
+		name: 'ship'
+	}
+	ship.advance()
+	assert true
+}

